### PR TITLE
Keys transformer

### PR DESCRIPTION
**Pull request by jeans11**
#58

### CODE REVIEW

**Comment by ikitommi** [comment]
Could you run the performance test before and after the change? in case there are no key-transformers defined, we should not get any performance penalty.

https://github.com/metosin/malli/blob/master/perf/malli/perf_test.cljc#L266

**Comment by jeans11** [comment]
Ok so performance issue found. I investigate.

**Comment by jeans11** [comment]
Performance should be ok now

File: src/malli/core.cljc [hunk 1]
```diff
@@ -233,10 +234,30 @@
...
+                                                        noop-transformer (fn [x] x)]
+                                                    (cond
+                                                      (and key-transformer value-transformer)
+                                                      [k [key-transformer value-transformer]]
+
+                                                      (and (not key-transformer) value-transformer)
+                                                      [k [noop-transformer value-transformer]]
+
+                                                      (and key-transformer (not value-transformer))
+                                                      [k [key-transformer noop-transformer]]))))
```
Comment (ikitommi) [inline src/malli/core.cljc]: I think all keys should be transformed, not just the ones defined in the schema. e.g. the map might be open and the value can have extra keys, which should be transformed too.

Given that, I think the key-transformation could be an extra step, just like the `map-transformer`. With this, the performance for not using that would be identical as before. If both the `map-transformer` & `key-transformer` are present, there could be a max of 3 steps. I think it's ok, can't think of a way doing it faster.

**Comment by jeans11** [comment]
Anything I can do to help this along?

**Comment by ikitommi** [comment]
looks great, sorry for taking so long. Thanks!

**Comment by ikitommi** [comment]
What do you think about https://github.com/metosin/malli/issues/91?

**Pull request merged**